### ISSUE TITLE
Could not reproduce the precision of ViT

### ISSUE THREAD

**Issue by amrta-coder** [Abhishek-Iyer1/emotion-classifcation-eeg-seed-ensemble#5]
@Abhishek-Iyer1 

Thank you very much for your extraordinary and fantastic work. I really learned a lot.

For the CNN, RNN, and Hybrid, I reproduced almost the same accuracy as you announced.
However, for ViT Modified, I tried a lot but could only obtain an accuracy of up to 88% in the training stage. Could you give me some hints on the training of ViT Modified?

By the way, some papers claimed that ViT could be used to raw data (rather than extracting characteristics but only pre-processing such as ICA processing) directly. However, I found that is even harder. Do you have some suggestions or ideas about this?

**Comment by wiltontsang** [comment]
@amrta-coder 
Hi！

I would like to know how to write the de_LDS function in the Matlab_dataset_correcting_code.m file. I have no idea about it，Could you please share with me, thank you！

Best

**Comment by Abhishek-Iyer1** [comment]
> @amrta-coder Hi！
> 
> I would like to know how to write the de_LDS function in the Matlab_dataset_correcting_code.m file. I have no idea about it，Could you please share with me, thank you！
> 
> Best

Hello! The `'de_LDS'` lines in the matlab dataset correcting code are not functions. The raw data had sets of 15 samples which I wanted to concatenate into one dataset array with all the raw data in order to preprocess. So what the whole file does, rather inefficiently, is load each of these sets and append the raw samples from those files into the data variable. `sprintf()` is used to concatenate `'de_LDS'` with the correct number in order to access the right data. Hope that helps!

**Comment by Abhishek-Iyer1** [comment]
> @Abhishek-Iyer1
> 
> Thank you very much for your extraordinary and fantastic work. I really learned a lot.
> 
> For the CNN, RNN, and Hybrid, I reproduced almost the same accuracy as you announced. However, for ViT Modified, I tried a lot but could only obtain an accuracy of up to 88% in the training stage. Could you give me some hints on the training of ViT Modified?
> 
> By the way, some papers claimed that ViT could be used to raw data (rather than extracting characteristics but only pre-processing such as ICA processing) directly. However, I found that is even harder. Do you have some suggestions or ideas about this?

Hey! First of all, apologies for the late reply. I am glad you found the work helpful and learned a lot. If working on research, please remember to cite our paper!

As for the issue in training the ViT Modified model, that is surprising as 88% is rather low in comparison to what CNN and LSTMs achieve. Are you using functions such as `ReduceLROnPlateau`? What other hyperparameters are you using to train the model? If I was to venture a guess, it could be that the model is stuck in a local optimum or that the learning rate has become too low by then to make any significant difference. 

With respect to your question on training ViT on Raw Data, yes that is something extraordinary that Vision Transformers are capable of unlike CNNs as using patches to create embeddings, they can train on higher resolution input data. However, when I was working on the project, the size of our raw data was too big and I would believe it would be even harder to train to get the correlation between raw data features and outputs right. So it could be challenging, but could be solved by using a GPU cluster if you have access to one and some reading up on how to train models on raw data.

**Comment by wiltontsang** [comment]
> > @amrta-coder Hi！
> > I would like to know how to write the de_LDS function in the Matlab_dataset_correcting_code.m file. I have no idea about it，Could you please share with me, thank you！
> > Best
> 
> Hello! The `'de_LDS'` lines in the matlab dataset correcting code are not functions. The raw data had sets of 15 samples which I wanted to concatenate into one dataset array with all the raw data in order to preprocess. So what the whole file does, rather inefficiently, is load each of these sets and append the raw samples from those files into the data variable. `sprintf()` is used to concatenate `'de_LDS'` with the correct number in order to access the right data. Hope that helps!

OK,I got it,and it does work,thank you very much!
